frequency_calculator.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+from typing import Annotated
+
+import typer
+from typing_extensions import Optional
+
+from characters_counter import count_character_frequency
+from logging_config import setup_logging
+
+app = typer.Typer()
+logger = setup_logging()
+
+
+@app.command()
+def calculate(
+    file: Annotated[Optional[Path], typer.Option("--file", "-f")] = None,
+    paragraph: Annotated[Optional[str], typer.Option("--paragraph", "-p")] = None,
+) -> None:
+    if file and paragraph:
```

Comment:
You can significantly reduce the number of if elifs in the code by following `guard clause` in python. Move negative checks to the beginning. Then check if you have file, if so -> call file logic. else call paragraph logic.